src/amake.f

```diff
@@ -637,10 +639,819 @@ SUBROUTINE MAKESURF(ISURF)
+      ! Strip right side
+      do idx_dim = 1,3
+       RLE2(idx_dim,idx_strip) = mesh(idx_dim,1,idx_y+1)
+      end do 
+      CHORD2(idx_strip) = sqrt((mesh(1,nx,idx_y+1)-mesh(1,1,idx_y+1))**2 
+     &                      + (mesh(3,nx,idx_y+1)-mesh(3,1,idx_y+1))**2)
+
+      ! Strip mid-point 
+      do idx_dim = 1,3
+       ! Since the strips are linear we can just interpolate
+       RLE(idx_dim,idx_strip) = (RLE1(idx_dim,idx_strip)
+     &   + RLE2(idx_dim,idx_strip))/2.
+       ! RLE(idx_dim,idx_strip) = (mesh(idx_dim,1,idx_y+1)+mesh(idx_dim,1,idx_y))/2 
+      end do 
+       ! Since the strips are linear we can just interpolate
+       CHORD(idx_strip) = (CHORD1(idx_strip)+CHORD2(idx_strip))/2.
+!       m1 = ((mesh(1,nx,idx_y+1)+mesh(1,nx,idx_y))/2) - 
+!      & ((mesh(1,1,idx_y+1)+mesh(1,1,idx_y))/2)
+!       m3 = ((mesh(3,nx,idx_y+1)+mesh(3,nx,idx_y))/2) - 
+!      & ((mesh(3,1,idx_y+1)+mesh(3,1,idx_y))/2)
+!       CHORD(idx_strip) = sqrt(m1**2 + m3**2)
+
+      ! Strip width (leading edge)
+      m2 = mesh(2,1,idx_y+1)-mesh(2,1,idx_y)
+      m3 = mesh(3,1,idx_y+1)-mesh(3,1,idx_y)
+      WSTRIP(idx_strip) = sqrt(m2**2 + m3**2)
+
+      ! Strip LE and TE sweep slopes
+      tanle(idx_strip) = (mesh(1,1,idx_y+1)-mesh(1,1,idx_y))
+     &                                            /WSTRIP(idx_strip)
+      tante(idx_strip) = (mesh(1,nx,idx_y+1)-mesh(1,nx,idx_y))
+     &                                            /WSTRIP(idx_strip)
+
+      ! Compute chord projections and strip twists
+      ! In AVL the AINCS are not interpolated. The chord projections are
+      ! So we have to replicate this effect.
+ 
+      ! Interpolation over the section: left, right, and midpoint
+      f1 = (mesh(2,1,idx_y)-mesh(2,1,iptl))/
+     & (mesh(2,1,iptr)-mesh(2,1,iptl))
+      f2 = (mesh(2,1,idx_y+1)-mesh(2,1,iptl))/
+     & (mesh(2,1,iptr)-mesh(2,1,iptl))
+      fc = (((mesh(2,1,idx_y+1)+mesh(2,1,idx_y))/2.) 
+     & -mesh(2,1,iptl))/(mesh(2,1,iptr)-mesh(2,1,iptl))
+
+      ! Strip left side incidence
+      CHSIN = CHSINL + f1*(CHSINR-CHSINL)
+      CHCOS = CHCOSL + f1*(CHCOSR-CHCOSL)
+      AINC1(idx_strip) = ATAN2(CHSIN,CHCOS)
+
+      ! Strip right side incidence
+      CHSIN = CHSINL + f2*(CHSINR-CHSINL)
+      CHCOS = CHCOSL + f2*(CHCOSR-CHCOSL)
+      AINC2(idx_strip) = ATAN2(CHSIN,CHCOS)
+
+      ! Strip mid-point incidence
+      CHSIN = CHSINL + fc*(CHSINR-CHSINL)
+      CHCOS = CHCOSL + fc*(CHCOSR-CHCOSL)
+      AINC(idx_strip) = ATAN2(CHSIN,CHCOS)
+
+      ! Set dv gains for incidence angles
+      ! Bring over the routine for this from Drela
+      DO N = 1, NDESIGN
+         CHSIN_G = (1.0-FC)*CHSINL_G(N) + FC*CHSINR_G(N)
+         CHCOS_G = (1.0-FC)*CHCOSL_G(N) + FC*CHCOSR_G(N)
+         AINC_G(idx_strip,N) = (CHCOS*CHSIN_G - CHSIN*CHCOS_G)
+     &                       / (CHSIN**2 + CHCOS**2)
+      ENDDO
+
+      ! We have to now setup any control surfaces we defined for this section
+      ! Bring over the routine for this from Drela
```

Comment:
See comment above about "from Drela". It would probably be more informative to say "from makesurf"